wandb/integration/ultralytics/bbox_utils.py

```diff
@@ -0,0 +1,184 @@
+
+
+def scale_bounding_box_to_original_image_shape(
+    box, resized_image_shape, original_image_shape, ratio_pad
+) -> List[int]:
+    """YOLOv8 resizes images during training and the label values are normalized based on this resized shape.
+
+    This function rescales the bounding box labels to the original
+    image shape.
+
+    Reference: https://github.com/ultralytics/ultralytics/blob/main/ultralytics/yolo/utils/callbacks/comet.py#L105
+    """
+    resized_image_height, resized_image_width = resized_image_shape
+    # Convert normalized xywh format predictions to xyxy in resized scale format
+    box = ops.xywhn2xyxy(box, h=resized_image_height, w=resized_image_width)
+    # Scale box predictions from resized image scale back to original image scale
+    box = ops.scale_boxes(resized_image_shape, box, original_image_shape, ratio_pad)
+    # # Convert bounding box format from xyxy to xywh for Comet logging
+    box = ops.xyxy2xywh(box)
+    return box.tolist()
+
+
+def get_ground_truth_bbox_annotations(
+    img_idx, image_path, batch, class_name_map=None
+) -> List[Dict[str, Any]]:
+    indices = batch["batch_idx"] == img_idx
+    bboxes = batch["bboxes"][indices]
+    cls_labels = batch["cls"][indices].squeeze(1).tolist()
+
+    class_name_map_reverse = {v: k for k, v in class_name_map.items()}
+
+    if len(bboxes) == 0:
+        wandb.termwarn(f"Image: {image_path} has no bounding boxes labels")
+        return None
+
+    cls_labels = batch["cls"][indices].squeeze(1).tolist()
+    if class_name_map:
+        cls_labels = [str(class_name_map[label]) for label in cls_labels]
+
+    original_image_shape = batch["ori_shape"][img_idx]
+    resized_image_shape = batch["resized_shape"][img_idx]
+    ratio_pad = batch["ratio_pad"][img_idx]
+
+    data = []
+    for box, label in zip(bboxes, cls_labels):
+        box = scale_bounding_box_to_original_image_shape(
+            box, resized_image_shape, original_image_shape, ratio_pad
+        )
+        data.append(
+            {
+                "position": {
+                    "middle": [int(box[0]), int(box[1])],
+                    "width": int(box[2]),
+                    "height": int(box[3]),
+                },
+                "domain": "pixel",
+                "class_id": class_name_map_reverse[label],
+                "box_caption": label,
+            }
+        )
+
+    return data
+
+
+def get_mean_confidence_map(
+    classes: List, confidence: List, class_id_to_label: Dict
+) -> Dict[str, float]:
+    confidence_map = {v: [] for _, v in class_id_to_label.items()}
+    for class_idx, confidence_value in zip(classes, confidence):
+        confidence_map[class_id_to_label[class_idx]].append(confidence_value)
+    for label, confidence_list in confidence_map.items():
+        if len(confidence_list) > 0:
+            confidence_map[label] = sum(confidence_list) / len(confidence_list)
+        else:
+            confidence_map[label] = 0
+    return confidence_map
```

Comment:
hmmm could you modify the code such that you don't change the map as you iterate over it?